// Copyright 2024 The gRPC authors.

syntax = "proto3";

package chaotic_good_frame;

message Settings {
    enum Features {
        UNSPECIFIED = 0;
        CHUNKING = 1;
    }

    repeated bytes connection_id = 1;

    bool data_channel = 2;

    uint32 alignment = 3;

    uint32 max_chunk_size = 4;

    repeated Features supported_features = 5;
}

message UnknownMetadata {
    string key = 1;
    bytes value = 2;
}

message ClientMetadata {
    optional string path = 1;
    optional string authority = 2;
    optional uint64 timeout_ms = 3;

    repeated UnknownMetadata unknown_metadata = 100;
}

message ServerMetadata {
    optional uint32 status = 1;
    optional bytes message = 2;

    repeated UnknownMetadata unknown_metadata = 100;
}

message BeginMessage {
    uint64 length = 1;
}
